dashboard/src/pages/api/run.ts

```diff
@@ -108,8 +108,18 @@ async function runSql(sql: string, dbAlias: string, username: string, password:
       host: baseConnConfig.host,
       ssl: baseConnConfig.extra.ssl,
     });
+    // Based on https://node-postgres.com/apis/client#error
+    connTemp.on('error', e => {
```

Comment:
Should we also pass the `res` object through to here and if this case is hit return an error to the front-end?

Reply:
I think we cannot since the listener is defined already in the `pg` lib. For the cases when this is hit due to `disconnect` killing open connections, it is ok not to do anything since that's exactly what disconnect is doing, cancel all ongoing paths.

Reply:
hmm, so the handler is being defined within the `runSql` function call so each call will get a new handler. This means that we could make `res` another argument to `runSql` and then use it here as part of the closure.

Whether that's actually necessary is something else; in this particular case, no, but if there's an error in the future due to some other issue the error could be useful.

Reply:
ohh I see your point, yes that could be useful. I could give it a try in this PR